### ISSUE TITLE
L037 False Positive with explicit ORDER BY expressions and dbt

### ISSUE THREAD

**Issue by WillAyd** [sqlfluff/sqlfluff#4194]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Using sqlfluff with dbt I had trouble getting the following SQL query to work:

```sql
SELECT * FROM a_table
ORDER BY a ASC, b DESC
```

sqlfluff would errantly throw and L037 violation of `Ambiguous order by clause. Order by clauses should specific order direction for ALL columns or NO columns`

### Expected Behaviour

No L037 should be triggered

### Observed Behaviour

I traced the issue back to this line in the code:

https://github.com/sqlfluff/sqlfluff/blob/edb79809d2d7e16933e977d72f7a7d7b491c8cbc/src/sqlfluff/rules/L037.py#L65

I _think_ this can be refactored to just be `if found_column_reference and ordering_reference:` to resolve. The issue _may_ have something to do with the fact that (at least using dbt) the parsed segments of the above query look something like:

```
<ColumnReferenceSegment: >
<Dedent: >
<TemplateLoop: >
<Indent: >
<WhitespaceSegment: ' '>
<KeywordSegment: 'ASC'>
```

So the `TemplateLoop` I think throws off the current logic

### How to reproduce

`sqlfluff --dialect redshift <the_file>` where <the_file> contains:

```sql
SELECT * FROM a_table
ORDER BY a ASC, b DESC
```

### Dialect

redshift

### Version

1.4.2

### Configuration

[sqlfluff]
templater = dbt

[sqlfluff:templater:dbt]
project_dir = <the_folder>

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
